src/index.js

```diff
@@ -40,7 +40,26 @@ function IpfsAPI (hostOrMultiaddr, port, opts) {
   cmds.send = requestAPI
   cmds.Buffer = Buffer
 
+  cmds.version(function (err, res) {
+    if (err) console.log(err)
+    var dotSplit = res.version.split('.')
+    if (dotSplit[0] === 0) {
+      if (dotSplit[1] < 4) {
+        throw new DaemonVersionError(res)
+      }
+    }
+  })
```

Comment:
since this check is async and the creation of the instance is sync, that throw will be an uncaught exception, you would have to make the creation of ipfs api async so that error could be passed on the callback.